core/store/src/trie/shard_tries.rs

```diff
@@ -627,4 +607,98 @@ mod test {
             shard_uid
         );
     }
+
+    #[test]
+    fn test_insert_delete_trie_cache() {
+        let store = create_test_store();
+        let trie_cache_config = TrieCacheConfig {
+            default_max_bytes: DEFAULT_SHARD_CACHE_TOTAL_SIZE_LIMIT,
+            per_shard_max_bytes: Default::default(),
+            shard_cache_deletions_queue_capacity: 0,
+        };
+        let trie_config = TrieConfig {
+            shard_cache_config: trie_cache_config.clone(),
+            view_shard_cache_config: trie_cache_config,
+            enable_receipt_prefetching: false,
+            sweat_prefetch_receivers: Vec::new(),
+            sweat_prefetch_senders: Vec::new(),
+        };
+        let shard_uids = Vec::from([ShardUId { shard_id: 0, version: 0 }]);
+        let shard_uid = *shard_uids.first().unwrap();
+
+        let trie = ShardTries::new(
+            store.clone(),
+            trie_config,
+            &shard_uids,
+            FlatStorageManager::test(store, &shard_uids, CryptoHash::default()),
+        );
+
+        let trie_caches = &trie.0.caches;
+        // Assert only one cache for one shard exists
+        assert_eq!(trie_caches.read().unwrap().len(), 1);
+        // Assert the shard uid is correct
+        assert!(trie_caches.read().unwrap().get(&shard_uid).is_some());
+
+        // Read from cache
+        let key = CryptoHash::from_str("32222222222233333333334444444444445555555777").unwrap();
```

Comment:
nit: let's make keys in tests looking simpler (like "0", "alice", etc), otherwise reader may think that there is something specific about these values. Same thought about the test below.